src/visGeometry/index.ts

```diff
@@ -650,6 +654,14 @@ class VisGeometry {
         return this.followObjectId;
     }
 
+    public getFollowObjectData(): AgentData {
```

Comment:
So this is already split! The existing method `getFollowObject` returns the id and `getFollowObjectData` returns the data, i could rename the existing function to make the difference clear.

I was following a pattern I observed where controller methods seem to be pre-defined/functional (play, zoom out, etc) and most cases of passing data to the front end used callbacks via viewport props. But if we put this in the controller then we could reduce the viewer code and make the front end responsible for how often it retrieves the data, currently its in the animate loop. I would be down to make that change.

Reply:
What I'm suggesting here is getObjectData(id) where you pass in an id and get the AgentData out. Instead of having a specific getFollowObjectData(), it would be getObjectData(getFollowObject())

Reply:
The question of whether things should be at the SimulariumController level is probably one of how the simularium-website needs to talk to the viewer.  I'm not really sure what is better, not having looked at the code recently.  Does website make a lot of direct calls on a SimulariumViewer or do they mostly go through Controller?

Reply:
The pattern of handling continuous updates is definitely something the viewer does already and the website does not. To handle it by calling a controller method in the front end, we would need to boot strap some approximation of the frame rate in requesting that data from the viewer.

After thinking about it I still think it makes the most sense to have it the way it is here: tell the viewer that when the scene updates the front end would like to know the data about a particular agent via a callback and then sit back and listen.